sound/soc/soc-topology.c

```diff
@@ -1137,8 +1137,6 @@ static int soc_tplg_dapm_widget_dmixer_create(struct soc_tplg *tplg, struct snd_
 	sm->min = le32_to_cpu(mc->min);
 	sm->invert = le32_to_cpu(mc->invert);
 	sm->platform_max = le32_to_cpu(mc->platform_max);
-	sm->dobj.index = tplg->index;
-	INIT_LIST_HEAD(&sm->dobj.list);
```

Comment:
it's very hard to connect the dots between the commit message and the removal of these lines. Better add information on where these assignments are done rather than just say 'no need to be redundant'.

Reply:
Thanks, would following be more clear:

Functions soc_tplg_dapm_widget_d[bytes|enum|mixer]_create() are called
from soc_tplg_dapm_widget_create() to register kcontrol.
soc_tplg_dapm_widget_create() registers widget as dynamic object itself.
When widget is removed soc_tplg_remove_widget() is called and removes
kcontrols. So there is no need to do dynamic object on kcontrols
themselves in this scenario, as they always have widget to which they
belong and which will clean them up.

Reply:
not much better haha...

"kcontrols do not need to be managed as topology dynamic objects with an index and a linked list. kcontrols are always associated with a widget which is already a topology dynamic object, thus all addition/removals of a widget will by construction manage the kcontrol"